packages/resin-event-log/test/resin-event-log.spec.js

```diff
@@ -44,17 +44,20 @@ function aggregateMock(mocks) {
 	}
 }
 
-function validateMixpanelQuery(event) {
+function validateMixpanelQuery(event, user) {
 	return function(queryObject) {
 		var data = queryObject.data
 		if (!data) return false
 
 		try {
 			data = JSON.parse(base64Decode(data))
+			// TODO why do we pass username instead on user.id as distinct_id?
```

Comment:
Because that's what we've historically been using as the distinct id everywhere. I don't think it matters which we use, since they're both unique identifiers, but it does need to be consistent everywhere, or we need to make sure we always alias them together. Easier to be consistent.

That's the historic position. For more current context, we're now (as of the UI deploy early this week I think), aliasing usernames to UUIDs at signup in the UI, so most users will really have UUID distinct ids. You can still track events and profile data against their usernames everywhere, but all that data will end up associated to the UUID profile instead.

This won't be 100% of users though, as we don't have proper handling for social signups (so they'll still just be usernames, and don't yet have funnel data), and it is possible to sign up without going through the UI at all, using the CLI or the API directly, where obviously we can't get funnel data anyways. Generally, using the username as a distinct id should end up doing the right thing everywhere.